pyneuroml/plot/Plot.py

```diff
@@ -0,0 +1,430 @@
+    :type verbose: boolean
+    :returns: matplotlib.axes.Axes object
+    """
+
+    logger.info("Generating plot: %s" % (title))
+
+    from matplotlib import pyplot as plt
+    from matplotlib import rcParams
+
+    rcParams.update({"font.size": font_size})
+
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+
+    plt.get_current_fig_manager().set_window_title(title)
+    if title_above_plot:
+        plt.title(title)
+
+    if xaxis:
+        plt.xlabel(xaxis)
+    if yaxis:
+        plt.ylabel(yaxis)
+
+    if grid:
+        plt.grid("on")
+
+    if logx:
+        ax.set_xscale("log")
+    if logy:
+        ax.set_yscale("log")
+
+    if bottom_left_spines_only:
+        ax.spines["right"].set_visible(False)
+        ax.spines["top"].set_visible(False)
+        ax.yaxis.set_ticks_position("left")
+        ax.xaxis.set_ticks_position("bottom")
+
+    if not show_xticklabels:
+        ax.set_xticklabels([])
+    if not show_yticklabels:
+        ax.set_yticklabels([])
+
+    for i in range(len(xvalues)):
+
+        linestyle = "-" if not linestyles else linestyles[i]
+        label = "" if not labels else labels[i]
+        marker = None if not markers else markers[i]
+        linewidth = 1 if not linewidths else linewidths[i]
+        markersize = 6 if not markersizes else markersizes[i]
+
+        if colors:
+            plt.plot(
+                xvalues[i],
+                yvalues[i],
+                "o",
+                color=colors[i],
+                marker=marker,
+                markersize=markersize,
+                linestyle=linestyle,
+                linewidth=linewidth,
+                label=label,
+            )
+        else:
+            plt.plot(
+                xvalues[i],
+                yvalues[i],
+                "o",
+                marker=marker,
+                markersize=markersize,
+                linestyle=linestyle,
+                linewidth=linewidth,
+                label=label,
+            )
+
+    if labels:
+        if legend_position == "right":
+            box = ax.get_position()
+            ax.set_position([box.x0, box.y0, box.width * 0.8, box.height])
+            # Put a legend to the right of the current axis
+            ax.legend(loc="center left", bbox_to_anchor=(1, 0.5))
+
+        else:
+            plt.legend(
+                loc="upper center",
+                bbox_to_anchor=(0.5, -0.05),
+                fancybox=True,
+                shadow=True,
+                ncol=cols_in_legend_box,
+            )
+
+    if xlim:
+        plt.xlim(xlim)
+    if ylim:
+        plt.ylim(ylim)
+
+    if save_figure_to:
+        logger.info(
+            "Saving image to %s of plot: %s" % (os.path.abspath(save_figure_to), title)
+        )
+        plt.savefig(save_figure_to, bbox_inches="tight")
+        logger.info("Saved image to %s of plot: %s" % (save_figure_to, title))
+
+    if show_plot_already:
+        plt.show()
+
+    return ax
+
+
+def generate_interactive_plot(
```

Comment:
@irahulsonkar @anujanegi @shayan823 : this is the new function (see the test file for example usage too).